source/features/quick-comment-edit.tsx

```diff
@@ -49,27 +50,13 @@ function addQuickEditButton(commentDropdown: HTMLDetailsElement, {signal}: Signa
 	);
 }
 
-export function canEditEveryComment(): boolean {
-	return elementExists([
-		// If you can lock conversations, you have write access
-		'.lock-toggle-link > .octicon-lock',
-
-		// Some pages like `isPRFiles` does not have a lock button
-		// These elements only exist if you commented on the page
-		'[aria-label^="You have been invited to collaborate"]',
-		'[aria-label^="You are the owner"]',
-		'[title^="You are a maintainer"]',
-		'[title^="You are a collaborator"]',
-	]) || pageDetect.canUserEditRepo();
-}
```

Comment:
My concern is that the check is no longer instantaneous :(

Maybe:

- cache the call
- also include DOM-based detections as a first check, like `elementExists(…) || await getViewerPermission() === 'something'`

Reply:
> Why did you choose v4uncached? The regular v4 should already cache by repo.

I think I just copy this code from `get-pr-info`. But `v4` it's better.